### ISSUE TITLE
Importer: Rename foursquare to swarm?

### ISSUE THREAD

**Issue by bradfitz** [perkeep/perkeep#991]
We probably want to rename the foursquare importer to Swarm, but without breaking people.

Probably only the UI should change to Swarm, but the underlying importer type can still be "foursquare" for compatibility. We could also change the directory on disk.

**Comment by lindner** [comment]
The interesting thing here is:

An importer should be able to handle the underlying service changing names,
owners and formats.
- That means that old data that was imported should continue to work
- New data that is imported should reflect the new branding
- Old data should potentially be visible with old branding or new.
- Importing old data and new data should map to equivalence using a
namespace or other mechanism.

For example Google Buzz content migrated from Buzz to Google+  URLs and
presentation changed, but content remained the same.

**Comment by bradfitz** [comment]
> - That means that old data that was imported should continue to work

Agreed. Don't think anybody would disagree with this.

> - New data that is imported should reflect the new branding

Ideally, but not a super high priority. Companies change their branding regularly. I care more about where I checked in last year rather than what their UI designers thought looked cool that particular day. But when things change a bunch, we should refresh our web UI to render it in the current-ish style.

> - Old data should potentially be visible with old branding or new.

I think this is a restatement of the first point?

> Importing old data and new data should map to equivalence using a namespace or other mechanism.

Agreed. And they do today: all of our imported objects are relative to the account node, and generally named in a path by their underlying IDs from the upstream services. And in cases where one importer broke down (e.g. the `picasa` importer broke down and wouldn't return more than 10,000 photos due the Picasa API), then we made sure the replacement importer (`gphotos`, using the Google Drive API) went out of its way to not create duplicate photo nodes.
